YanZhylavy/HW 8/Practical task 8.2.py

```diff
@@ -0,0 +1,34 @@
+password = input('Enter password: ')
```

Comment:
optimized version of your implementation

password = input('Enter password: ')
criteria = {'digits': False, 'uppers': False, 'lowers': False, 'sym': False, 'length': False}
symbols = ['#', '@', '$']

for i in password:
    if i.isdigit():
        criteria['digits'] = True
    elif i.isupper():
        criteria['uppers'] = True
    elif i.islower():
        criteria['lowers'] = True
    elif i in symbols:
        criteria['sym'] = True

if 6 <= len(password) <= 16:
    criteria['length'] = True

if all(criteria.values()):
    print('Password is valid')
else:
    print('Password is invalid')